.github/workflows/test_and_deploy.yml

```
@@ -0,0 +1,137 @@
+        run: pipx install cibuildwheel==2.11.4
+      - id: set-matrix
+        run: |
+          MATRIX=$(
+            {
+              cibuildwheel --print-build-identifiers --platform linux \
+              | jq -nRc '{"only": inputs, "os": "ubuntu-20.04"}' \
+              && cibuildwheel --print-build-identifiers --platform macos \
+              | jq -nRc '{"only": inputs, "os": "macos-11"}' \
+              && cibuildwheel --print-build-identifiers --platform windows \
+              | jq -nRc '{"only": inputs, "os": "windows-2019"}'
+            } | jq -sc
+          )
+          echo "include=$MATRIX" | tee -a $GITHUB_OUTPUT
+        env:
+          CIBW_ARCHS_LINUX: "x86_64 aarch64"
+          CIBW_ARCHS_MACOS: "x86_64 arm64"
+          CIBW_ARCHS_WINDOWS: "x86 AMD64"
+          # Skip musllinux because it takes too long to compile on GHA
+          # since it is emulated. (6+ hours)
+          # *note* most of the build time is actually numpy for musllinux
+          CIBW_SKIP: "*musllinux* *-manylinux_i686 *-musllinux_i686 *-win32 pp*"
+
+  test-biotite-and-build-wheels:
+    name: "Build & Test"
+    needs: "generate-wheels-matrix"
+    strategy:
+      matrix:
+        include: ${{ fromJson(needs.generate-wheels-matrix.outputs.include) }}
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up QEMU
+        if: runner.os == 'Linux'
+        uses: docker/setup-qemu-action@v2
+        with:
+          platforms: all
+
+      - name: Build & (optionally) Test wheels
+        # MAKE SURE THIS STAYS IN SYNC WITH THE UPPER pipx call to cibuildwheel
+        uses: pypa/cibuildwheel@v2.11.4
+        with:
+          only: ${{ matrix.only }}
+
+      - uses: actions/upload-artifact@v3
+        with:
+          path: ./wheelhouse/*.whl
+  
+
+  test-muscle5:
+    name: Testing Biotite's interface for Muscle 5
+
+    runs-on: ubuntu-20.04
+    defaults:
+      run:
+        shell: bash -l {0}
+
+    steps:
+      - uses: actions/checkout@v3
+      - uses: conda-incubator/setup-miniconda@v2
+        with:
+          activate-environment: biotite-dev
+          python-version: 3.9
+          channels: conda-forge,defaults
+      - name: Installing dependencies
+        run: conda install -c bioconda "muscle=5"
+      - name: Building distribution
+        run: pip wheel --no-deps -w dist .
+      - name: Installing distribution
+        run: pip install .//dist//*.whl pytest
+      - name: Testing code
+        run: mkdir tmp && cd tmp && pytest ..//tests//application//test_msa.py --durations=10 && cd .. && rm -r tmp
+
+  make_sdist:
+    name: Make SDist
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+        submodules: true
+    - name: Build SDist
+      run: pipx run build --sdist
+    - uses: actions/upload-artifact@v3
+      with:
+        path: dist/*.tar.gz
+
+  upload_to_pypi_if_release:
```

I think a verbose name would be nice here to improve the graph of the action.
```suggestion
  upload_to_pypi_if_release:
    name: Upload to PyPI
```